### ISSUE TITLE
Create Class edgeStorage

### ISSUE THREAD

**Issue by MuellMark** [MuellMark/GraphLabeling2023#11]
I originally didn't want this to be it's own issue, but this will take some work to get working and I don't want to clog up #10, so anything relate to the class will go here.

**Comment by MuellMark** [comment]
The class now has most, if not all, features needed, so I can now start redoing the generation code. I made some choices that may not be obvious in the creation of the class so I want to list them here: 

- The size of the object refers to how many vertices we're considering, not the actual size. The size is one more since I'm not using 0 to make it easier to read
- All of the get methods had to be copies to avoid the pointer issues I've been having
- The copy method calls on all of these these methods to also avoid point issue
- There are several add methods, one for adding in one element (both for in or out), one for adding a pair (main one I'll be using), and one for adding a whole array list, mostly for the copy method.

From here, I'm going to be working on the generation code, which should be much easier to do now that I've already started it, and the class should be working
